### DISCUSSION TITLE
Common chunks in serverless page bundles

### DISCUSSION THREAD

**Discussion by danielcondemarin** [vercel/next.js#10456] [Help] [answered]
As I understand it there is quite a bit of overlap / duplicated code between each serverless page bundle, where each base file at least includes `react` and `react-dom`. Adding external libraries also leads to an increase in each page bundle size.

This does not pose a problem when you deploy one Lambda Fn per serverless page bundle (like Now and [serverless-nextjs-plugin](https://github.com/danielcondemarin/serverless-next.js/tree/master/packages/serverless-nextjs-plugin) do. 

However for [this project](https://github.com/danielcondemarin/serverless-next.js) I've been maintaining for a while now is not possible to deploy one Lambda Fn. per page as it uses Lambda@Edge behind the scenes where doing this isn't possible. 
An example of the issue can be found in https://github.com/danielcondemarin/serverless-next.js/issues/236 where a user hit the limit of how big the Lambda artifact can be. 

Is it possible to output common chunks between sls page bundles? This would considerably reduce duplication. Considering the next framework is pushing so much for serverless this change would give library maintainers like myself the flexibility to take even further what's possible in this area.

I'm more than happy to look into it myself, all I'd need would be a bit of initial guidance 🙌

**Comment by timneutkens** [comment, marked as answer]
> Is it possible to output common chunks between sls page bundles?

Not as of right now with webpack. splitChunks is not supported for server-side compilation.
